l10n_ch_invoice_reports/models/mail_template.py

```diff
@@ -34,14 +29,11 @@ def _should_print_invoice_attachment(self):
         ):
             return True
 
-    def generate_invoice_attachment(self, template, res_id, related_record):
-        self._render_template(template.report_name, template.model, res_id)
+    def generate_invoice_attachment(self, invoice):
         # force translation of attachment's name according to partner's lang
-        self = self.with_context(lang=related_record.partner_id.lang)
-        report_name = _("invoice_%s_with_payslip.pdf") % related_record.name.replace(
-            "/", "_"
-        )
+        self = self.with_context(lang=invoice.partner_id.lang)
+        report_name = _("invoice_%s_with_payslip.pdf") % invoice.name.replace("/", "_")
         report_xml_id = self.report_template.xml_id
-        report_pdf = self.env.ref(report_xml_id).render_qweb_pdf([res_id])[0]
+        report_pdf = self.env.ref(report_xml_id).render_qweb_pdf([invoice.id])[0]
```

Comment:
Am I wrong or `self.report_template` IS already the `ir.actions.report` you need? 
If that's true, you don't need to use its xid to find it again.